HandEyeCalibration/Transform/Transform.cpp

```diff
@@ -0,0 +1,330 @@
+/*
+Convenience functions for pose and co-ordinate system transformations.
+*/
+
+#include <Eigen/Core>
+#include <Eigen/Dense>
+#include <Eigen/Geometry>
+
+#include <opencv2/core/core.hpp>
+
+#include <cmath>
+#include <iostream>
+
+#define _USE_MATH_DEFINES
+
+namespace TransformUtils
+{
+    enum Rotation
+    {
+        XYZ_INTRINSIC,
+        XYZ_EXTRINSIC,
+        ZYX_INTRINSIC,
+        ZYX_EXTRINSIC
+    };
+
+    struct Rotation_s
+    {
+        Rotation rot;
+        std::string short_desc;
+        std::string rpy_name;
+    };
+
+    cv::Mat eigen2cv(const Eigen::MatrixXd &eigenMat)
+    {
+        cv::Mat cvMat(static_cast<int>(eigenMat.rows()), static_cast<int>(eigenMat.cols()), CV_64FC1, cv::Scalar(0));
+
+        for(int i = 0; i < cvMat.rows; i++)
+        {
+            for(int j = 0; j < cvMat.cols; j++)
+            {
+                cvMat.at<double>(i, j) = eigenMat(i, j);
+            }
+        }
+
+        return cvMat;
+    }
+
+    Eigen::MatrixXd cv2eigen(const cv::Mat &cvMat)
```

Comment:
Christian is using cvToEigen (see below) so I suggest going for the same.

https://github.com/zivid/cpp-extra-samples/blob/master/HandEyeCalibration/UtilizeEyeInHandCalibration/UtilizeEyeInHandCalibration.cpp